data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters.py

```diff
@@ -1,131 +1,262 @@
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585795 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585802 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585820 AND (value_as_number < 0 OR value_as_number > 255)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585864 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585870 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585873 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1586159 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1586162 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    ELSE value_as_concept_id
+END AS
+    value_as_concept_id,
+    qualifier_concept_id,
+    unit_concept_id,
+    provider_id,
+    visit_occurrence_id,
+    observation_source_value,
+    observation_source_concept_id,
+    unit_source_value,
+    qualifier_source_value,
+    value_source_concept_id,
+    value_source_value,
+    questionnaire_response_id
 FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585864 AND (value_as_number < 0 OR value_as_number > 99)
+    {{project}}.{{dataset}}.observation""")
 
-UNION ALL
 
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585870 AND (value_as_number < 0 OR value_as_number > 99)
+class CleanPPINumericFieldsUsingParameters(BaseCleaningRule):
+    """
+    Apply value ranges to ensure that values are reasonable and to minimize the likelihood
+    of sensitive information (like phone numbers) within the free text fields.
+    """
 
-UNION ALL
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
 
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585873 AND (value_as_number < 0 OR value_as_number > 99)
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Sets value_as_number to NULL and value_as_concept_id and observation_type_concept_id '
+            'to new AOU custom concept 2000000010')
+        super().__init__(issue_numbers=['DC-827', 'DC-502', 'DC-487'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.RDR],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
```

Comment:
affected_tables=['observation'] shoul dbe added to the init method.

Reply:
Added